content/chainguard/chainguard-enforce/authentication/identity-examples/enforce-aws-identity.md

```diff
@@ -0,0 +1,359 @@
+
+The file will consist of the following content.
+
+```
+terraform {
+  required_providers {
+    chainguard = {
+      source = "chainguard/chainguard"
+    }
+    aws = {
+      source = "hashicorp/aws"
+    }
+  }
+}
+
+provider "chainguard" {}
+
+provider "aws" {}
+```
+
+This is a fairly barebones Terraform configuration file, but we will define the rest of the resources in the other two files. In `main.tf`, we declare and initialize the Chainguard Terraform provider.
+
+Next, you can create the `sample.tf` file.
+
+
+### `sample.tf`
+
+`sample.tf` will create a couple of structures that will help us test out the identity in a workflow.
+
+This Terraform configuration consists of two main parts. The first part of the file will contain the following lines.
+
+```
+resource "chainguard_group" "example-group" {
+  name   	 = "example-group"
+  description = <<EOF
+    This group simulates an end-user group, which the AWS role identity
+    can interact with via the identity in aws.tf.
+  EOF
+}
+```
+
+This section creates a Chainguard Enforce IAM group named `example-group`, as well as a description of the group. This will serve as some data for the identity — which will be created by the `aws.tf` file — to access when we test it out later on.
+
+The next section contains these lines, which create a sample policy and apply it to the `example-group` group created in the previous section.
+
+```
+resource "chainguard_policy" "cgr-trusted" {
+  parent_id   = chainguard_group.user-group.id
+  document = jsonencode({
+    apiVersion = "policy.sigstore.dev/v1beta1"
+    kind  	 = "ClusterImagePolicy"
+    metadata = {
+      name = "trust-any-cgr"
+    }
+    spec = {
+      images = [{
+        glob = "cgr.dev/**"
+      }]
+      authorities = [{
+        static = {
+          action = "pass"
+        }
+      }]
+    }
+  })
+}
+```
+
+This policy trusts everything coming from the [Chainguard Registry](/chainguard/chainguard-images/registry/overview/). Because this policy is broadly permissive, it wouldn't be practical or secure to use in a real-world scenario. Like the example group, this policy serves as some data for the AWS role to inspect after it assumes the Chainguard identity.
+
+Now you can move on to creating the last of our Terraform configuration files, `aws.tf`.
+
+### `aws.tf`
+
+The `aws.tf` file is what will actually create the identity for your AWS role to assume. The file will consist of four sections, which we'll go over one by one.
+
+The first section creates the identity itself.
+
+```
+resource "chainguard_identity" "aws" {
+  parent_id   = var.group
+  name        = "aws-identity"
+  description = "Identity for AWS role to assume"
+
+  aws_identity {
+    aws_account         = "[AWS-ACCOUNT-ID]"
+    aws_user_id_pattern = "^AROA(.*):[AWS-ROLE-SESSION-NAME]$"
+
+    // NB: This role will be assumed so can't use the role ARN directly. We must used the ARN of the assumed role
```

Comment:
Could you move this detail to the text below? A lot of readers (including me) tend to ignore comments in code blocks like this